frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -299,135 +306,166 @@ const Diagrams = () => {
-                    previousBlocks: previousBlocks.filter(
-                      (previousBlock) => previousBlock !== sourceId,
-                    ),
-                  }),
-                });
-              },
-            },
-          );
-        } else if (
-          link?.sourcePort.options.label === BlockPorts.attachmentOutPort
-        ) {
-          // Attached / AttachedTo Link Delete
-          await updateBlock(
-            {
-              id: sourceId,
-              params: {
-                attachedBlock: null,
-              },
-            },
-            {
-              onSuccess() {
-                updateCachedBlock({
-                  id: targetId,
-                  preprocess: (oldData) => ({
-                    ...oldData,
-                    attachedToBlock: null,
-                  }),
-                });
-              },
-            },
-          );
-        }
-      } else {
-        // Block Delete Case
-        const ids = id.includes(",") ? id.split(",") : [id];
-        const deletePromises = ids.map((id) => {
-          const block = getBlockFromCache(id);
-
-          return deleteBlock(id, {
-            onSuccess() {
-              // Update all linked blocks to remove any reference to the deleted block
-              [
-                ...(block?.nextBlocks || []),
-                ...(block?.previousBlocks || []),
-                ...(block?.attachedBlock ? [block.attachedBlock] : []),
-                ...(block?.attachedToBlock ? [block.attachedToBlock] : []),
-              ]
-                .map((bid) => getBlockFromCache(bid))
-                .filter((b) => !!b)
-                .forEach((b) => {
-                  updateCachedBlock({
-                    id: b.id,
-                    payload: {
-                      ...b,
-                      nextBlocks: b.nextBlocks?.filter(
-                        (nextBlockId) => nextBlockId !== id,
-                      ),
-                      previousBlocks: b.previousBlocks?.filter(
-                        (previousBlockId) => previousBlockId !== id,
-                      ),
-                      attachedBlock:
-                        b.attachedBlock === id ? undefined : b.attachedBlock,
-                      attachedToBlock:
-                        b.attachedToBlock === id
-                          ? undefined
-                          : b.attachedToBlock,
-                    },
-                    strategy: "overwrite",
-                  });
-                });
+    const ids = deleteDialogCtl?.data;
 
-              deleteCachedBlock(id);
-            },
-          });
-        });
-
-        await Promise.all(deletePromises);
-      }
+    if (!ids || ids?.length === 0) {
+      return;
+    }
+    const isLink = ids[0].length === 36;
 
-      deleteCallbackRef.current?.();
-      deleteCallbackRef.current = () => {};
-      deleteDialogCtl.closeDialog();
+    if (isLink) {
+      await handleLinkDeletion(ids[0]);
```

Comment:
Currently, the implemented logic only covers deleting one Link, while the user could select more than one link at the same time or could select many blocks and many links at the same time and trigger the action to delete them.

Reply:
I discussed it with @marrouchi and we decided on keeping the same logic for now and in the future update it

Reply:
@medchedli We will handle this issue in a seperate PR, can you open an issue about blocks VS links deletion ?